app/controllers/account_reset/delete_account_controller.rb

```diff
@@ -7,8 +7,10 @@ class DeleteAccountController < ApplicationController
     def show; end
 
     def delete
-      analytics.track_event(Analytics::ACCOUNT_RESET, event: :delete, token_valid: true)
-      email = reset_session_and_set_email
+      user = @account_reset_request.user
+      analytics.track_event(Analytics::ACCOUNT_RESET,
```

Comment:
Same comment here about using a Service Object whose only purpose here is to process the deletion and validate the token, and returns a FormResponse object, much like the `EmailConfirmationTokenValidator` I linked to above.

Reply:
The AccountReset service is currently doing too much. I would break it up into smaller services that each have a single responsibility.

Reply:
The general rule is that a Service Object should only have one public method.

Reply:
I actually love that rule.  Now all the rails community needs to do is apply it to controllers too.  That was partly how in another life I was able to ditch the routes file and breakup the Rails monolith.  Hence why you'll notice my controller names sometimes slip in a verb (which is not appropriate in this codebase).  We can refactor that service object separately because I do think it's the right way to go.